Mapper testing error with contributor installation options

<!-- Instructions For Filing a Bug: https://github.com/giotto-learn/giotto-learn/blob/master/CONTRIBUTING.rst -->

#### Description
<!-- Example: Joblib Error thrown when calling fit on VietorisRipsPersistence 
-->
`ModuleNotFoundError: No module named 'hypothesis'`, on 3 mapper tests (`test_filter`, `test_cover`, `test_nerve`), when running `pytest giotto`.

#### Steps/Code to Reproduce
<!--
If the code is too long, feel free to put it in a public gist and link
it in the issue: https://gist.github.com
-->
I have installed Giotto-learn, following the *Developer* installation steps:
```
git clone https://github.com/giotto-ai/giotto-learn.git
cd giotto-learn
pip install -e .
pytest giotto
```
On the other hand, import giotto` succeeds.

#### Expected Results
Tests passed, without ModuleNotFoundErrors.

#### Actual Results
<!-- Please paste or specifically describe the actual output or traceback. -->
1. ERROR collecting giotto/mapper/tests/test_cover.py  (...) Traceback:
giotto/mapper/tests/test_cover.py:2: in <module>
    from hypothesis import given
E   ModuleNotFoundError: No module named 'hypothesis'
2. ERROR collecting giotto/mapper/tests/test_filter.py (...) Traceback:
giotto/mapper/tests/test_filter.py:2: in <module>
    from hypothesis import given
E   ModuleNotFoundError: No module named 'hypothesis'
3. ERROR collecting giotto/mapper/tests/test_nerve.py (...) Traceback:
giotto/mapper/tests/test_nerve.py:2: in <module>
    from hypothesis import given
E   ModuleNotFoundError: No module named 'hypothesis'

#### Versions
Python 3.7.3
NumPy 1.18.1
SciPy 1.4.1
joblib 0.14.1
Scikit-Learn 0.22
giotto-Learn 0.1.3
[bugreport_ModuleNotFound_hypothesis.txt](https://github.com/giotto-ai/giotto-learn/files/4054424/bugreport_ModuleNotFound_hypothesis.txt)

--- a/README.rst
+++ b/README.rst
@@ -110,7 +110,7 @@ To install:
 .. code-block:: bash
 
    cd giotto-learn
-   pip install -e .
+   pip install -e ".[tests, doc]"
 
 From there any change in the library files will be immediately available on your machine.
 

